### PR TITLE
Use Set method insted of On and Off method

### LINKED ISSUE #2769: Use Set method insted of On and Off method
### Describe what maintenance you would like added.

Some code uses VTK's `*On()` and `*Off()` methods (ex: `QuadsOn()` and `QuadsOff()` in `vtkTessellatedBoxSource` class). These generally implement a method called `Set*()` (ex: `SetQuads()`), which can be refactored to avoid using if statements.

_Inspired by @adeak in https://github.com/pyvista/pyvista/pull/2767#discussion_r894997566_

### Links to source code.

You can find it by running `grep "On()" ./pyvista/*/*.py` in root path. For example:

https://github.com/pyvista/pyvista/blob/fc9cea791d3d8aa873ba3123ee9d1c0e8d065faa/pyvista/utilities/geometric_objects.py#L675-L678

### Pseudocode or Screenshots

```python 
 src.SetQuads(quads)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by RobPasMue**
### Overview

Resolves #2769 

### Details

Only conditional statements in which `On()` and `Off()` statements were used have been refactored.

### CODE REVIEW

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Good catch. I have some suggestions. It is better to have an explicit code in "else" case.

File: pyvista/plotting/scalar_bars.py [hunk 1]
```diff
@@ -411,10 +411,8 @@ def add_scalar_bar(
         if n_colors is not None:
             scalar_bar.SetMaximumNumberOfColors(n_colors)
 
-        if n_labels < 1:
-            scalar_bar.DrawTickLabelsOff()
-        else:
-            scalar_bar.DrawTickLabelsOn()
+        if not n_labels < 1:
+            scalar_bar.SetDrawTickLabels(True)
```
Comment (RobPasMue) [inline pyvista/plotting/scalar_bars.py:415]:
```suggestion
        if n_labels < 1:
            scalar_bar.SetDrawTickLabels(False)
        else:
            scalar_bar.SetDrawTickLabels(True)
```

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Comment by RobPasMue** [comment]
Batch suggestions applied are causing test failures. Looking into issue

**Comment by RobPasMue** [comment]
@tkoyama010 problem solved. Ready to merge.

File: pyvista/plotting/renderer.py [hunk 2]
```diff
@@ -2979,10 +2979,10 @@ def add_legend(
             self._legend.SetPosition(x, y)
             self._legend.SetPosition2(size[0], size[1])
 
-        if bcolor is None:
-            self._legend.UseBackgroundOff()
+        if bcolor is not None:
+            self._legend.SetUseBackground(True)
```
Comment (RobPasMue) [inline pyvista/plotting/renderer.py:2983]:
```suggestion
        if bcolor is None:
            self._legend.SetUseBackground(False)
```

File: pyvista/plotting/renderer.py [hunk 3]
```diff
@@ -2979,10 +2979,10 @@ def add_legend(
...
         else:
-            self._legend.UseBackgroundOn()
+            self._legend.SetUseBackground(False)
```
Comment (RobPasMue) [inline pyvista/plotting/renderer.py:2985]:
```suggestion
            self._legend.SetUseBackground(True)
```

**Review by akaszynski — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**